subsys/net/lib/coap/coap.c

```diff
@@ -217,6 +217,14 @@ static int parse_option(const struct coap_packet *cpkt,
 	}
 
 	if (option) {
+		if (len > sizeof(option->value)) {
+			NET_ERR("Incoming COAP option length is %u!\n"
```

Comment:
suggestion: what about single line error message and add more comments in the code? Two advantages with this approach 1) User can easily grep "error message" and 2) Less binary size.

Reply:
Sounds good.  How about a comment like this above the if check:
/*
 * Make sure the option data will fit into the value field of coap_option.
 * NOTE: To expand the size of the value field set:
 * CONFIG_COAP_EXTENDED_OPTIONS_LEN=y
 * CONFIG_COAP_EXTENDED_OPTIONS_LEN_VALUE=<new size>
 */

Then error can be something like:
"%u is > coap_option->value size(%u)!"

Reply:
Ok, good.